lib/scala/project-manager/src/main/scala/org/enso/projectmanager/infrastructure/http/ProjectsEndpoint.scala

```diff
@@ -0,0 +1,93 @@
+package org.enso.projectmanager.infrastructure.http
+
+import akka.http.scaladsl.model.{
+  ContentType,
+  HttpEntity,
+  MediaType,
+  StatusCodes
+}
+import akka.http.scaladsl.server.Directives._
+import akka.http.scaladsl.server.Route
+import com.typesafe.scalalogging.LazyLogging
+import org.enso.jsonrpc.Endpoint
+import org.enso.pkg.archive.EnsoProjectArchive
+import org.enso.projectmanager.control.core.CovariantFlatMap
+import org.enso.projectmanager.control.core.syntax._
+import org.enso.projectmanager.control.effect.Exec
+import org.enso.projectmanager.infrastructure.repository.{
+  ProjectRepository,
+  ProjectRepositoryFailure
+}
+
+import java.util.UUID
+
+import scala.concurrent.Future
+import scala.util.{Failure, Success}
+
+final class ProjectsEndpoint[
+  F[+_, +_]: Exec: CovariantFlatMap
+](repo: ProjectRepository[F])
+    extends Endpoint
+    with LazyLogging {
+
+  /** @inheritdoc */
+  override def route: Route =
+    projectsEndpoint
+
+  private val projectsEndpoint = {
+    path("projects" / JavaUUID / "enso-project") { projectId =>
+      get {
+        getEnsoProject(projectId)
+      }
+    }
+  }
+
+  private def getEnsoProject(projectId: UUID): Route = {
+    onComplete(buildEnsoArchive(projectId)) {
+      case Failure(err) =>
+        logger.error(
+          "Failure when building an enso-project archive [{}].",
+          projectId,
+          err
+        )
+        complete(StatusCodes.InternalServerError)
+      case Success(Left(err)) =>
+        logger.error(
+          "Failed to build an enso-project archive [{}]. {}",
+          projectId,
+          err
+        )
+        complete(StatusCodes.InternalServerError)
+      case Success(Right(None)) =>
+        complete(StatusCodes.NotFound)
+      case Success(Right(Some(archive))) =>
+        complete(
+          HttpEntity(
```

Comment:
Shouldn't we also write a test for this endpoint?

Reply:
All the tricky logic is about the archive building. This one is pretty straightforward. I'll see if I can mock the project repository and add a test then